compiler/passes/normalize.cpp

```diff
@@ -2616,6 +2625,63 @@ static void hack_resolve_types(ArgSymbol* arg) {
   }
 }
 
+/************************************* | **************************************
+*                                                                             *
+* We cannot export a function with an array argument directly, due to the     *
+* type being considered generic and due to other languages not understanding  *
+* our array structure, as well as our normal array structure assuming it has  *
+* control over the memory involved.  Instead, build a wrapper for the         *
+* function.  This wrapper will be modified to take in a c_ptr of the right    *
+* type and the size of the array, in addition to its other arguments.         *
```

Comment:
I thought the wrapper was using the runtime-defined array struct?

Reply:
This is what I get for writing comments early, I should be better about updating them

Reply:
There's no judgement here, just pointing it out for fixing. I figured that was what happened & expect it will happen again (to any / all of us).

Reply:
Thanks :)